### PR TITLE
Fixed Exception Handling issues for Delete and Search, made code improvements

### PR DESCRIPTION
**Pull request by sethlxk**

### CODE REVIEW

File: src/main/java/tp/DeleteAppointmentCommand.java [hunk 1]
```diff
@@ -17,8 +17,13 @@ public String execute(DoctorList doctorList, PatientList patientList,
                           PatientStorage patientStorage,
                           AppointmentStorage appointmentStorage) throws IHospitalException {
         Appointment curr = appointmentList.deleteAppointment(index);
-        return String.format(boundary + "Noted. I've removed this appointment:\n" + curr
-                                     + "\n" + "Now you have " + (appointmentList.getSize())
-                                     + " appointments recorded in the system." + System.lineSeparator() + boundary);
+        if (curr == null){
+            return String.format("The index " + index + " is not valid in the appointment list");
+        }
+        else {
```
Comment (cczhouqi) [inline src/main/java/tp/DeleteAppointmentCommand.java:23]: Remember to put "else" after the close curly bracket of your "if" statement to follow the consistent style :D

File: src/main/java/tp/DeleteDoctorCommand.java [hunk 2]
```diff
@@ -15,8 +15,13 @@ public String execute(DoctorList doctorList, PatientList patientList,
                         PatientStorage patientStorage,
                         AppointmentStorage appointmentStorage) throws IHospitalException {
         Doctor curr = doctorList.deleteDoctor(index);
-        return String.format(boundary + "Noted. I've removed this doctor:" + curr
-                + "\n" + "Now you have " + doctorList.getSize()
-                + " doctors in the system." + System.lineSeparator() + boundary);
+        if (curr == null){
```
Comment (cczhouqi) [inline src/main/java/tp/DeleteDoctorCommand.java:18]: Screen Shot 2022-04-05 at 11 53 02 PM
You will need to resolve the conflicts before merging.

Comment (Demonshaha) [reply src/main/java/tp/DeleteDoctorCommand.java:18]: should pass the checkstyles

Comment (Demonshaha) [reply src/main/java/tp/DeleteDoctorCommand.java:18]: should resolvethe merge conflict

**Review by cczhouqi — APPROVED** [review]

**Pull request merged**